### PR TITLE
📖 .github: improve kubernetes bump issue template

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

After working on:

- https://github.com/kubernetes/test-infra/pull/30347
- #9160

this fixes or clarifies some points in the template for the test jobs and makes it more accurate. Also updates the prior art links.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/area documentation

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/cc @kubernetes-sigs/cluster-api-release-team

**Comment by chrischdi** [comment]
Cross-checking if there is a general issue on 1-27-latest

/test pull-cluster-api-e2e-workload-upgrade-1-27-latest-main

File: .github/ISSUE_TEMPLATE/kubernetes_bump.md [hunk 1]
```diff
@@ -26,24 +26,32 @@ changes should be cherry-picked to all release series that will support the new
     * `test/*`: search for occurrences of the previous Kubernetes version
     * `Tiltfile`
   * Ensure the latest available kind version is used (including the latest images for this kind release)
+    * Add new images in the [kind mapper.go](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/infrastructure/kind/mapper.go#L79)
```
Comment (furkatgofurov7) [inline .github/ISSUE_TEMPLATE/kubernetes_bump.md:29]: can we also clarify please where to get kind image sha from?

File: .github/ISSUE_TEMPLATE/kubernetes_bump.md [hunk 2]
```diff
@@ -26,24 +26,32 @@ changes should be cherry-picked to all release series that will support the new
...
+    * Set new default image for the [test framework](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/framework/bootstrap/kind_provider.go#L40)
   * Verify the quickstart manually
-  * Prior art: #7156
-  * bump `InitWithKubernetesVersion` and `WorkloadKubernetesVersion` in `clusterctl_upgrade_test.go`
+  * Bump `InitWithKubernetesVersion` and `WorkloadKubernetesVersion` in `clusterctl_upgrade_test.go`
+    * Note: Do not bump for upgrades which are using older clusterctl versions.
```
Comment (furkatgofurov7) [inline .github/ISSUE_TEMPLATE/kubernetes_bump.md:33]:
```suggestion
    * Note: Do not bump for upgrades using older clusterctl versions.
```

**Review by furkatgofurov7 — COMMENTED** [review]
Looks good, few suggestions below:

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

Thanks!

File: .github/ISSUE_TEMPLATE/kubernetes_bump.md [hunk 3]
```diff
@@ -26,24 +26,33 @@ changes should be cherry-picked to all release series that will support the new
...
     * `Tiltfile`
   * Ensure the latest available kind version is used (including the latest images for this kind release)
+    * Add new images in the [kind mapper.go](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/infrastructure/kind/mapper.go#L79).
+      * See the [kind releases page](https://github.com/kubernetes-sigs/kind/releases) for the list of released images.
+    * Set new default image for the [test framework](https://github.com/kubernetes-sigs/cluster-api/blob/48ae58e51f9723ab7b9635d0e05ee54c4843707a/test/framework/bootstrap/kind_provider.go#L40)
   * Verify the quickstart manually
-  * Prior art: #7156
-  * bump `InitWithKubernetesVersion` and `WorkloadKubernetesVersion` in `clusterctl_upgrade_test.go`
+  * Bump `InitWithKubernetesVersion` and `WorkloadKubernetesVersion` in `clusterctl_upgrade_test.go`
+    * Note: Do not bump for upgrades using older clusterctl versions.
```
Comment (sbueringer) [inline .github/ISSUE_TEMPLATE/kubernetes_bump.md:34]:
```suggestion
    * Note: Only bump for Cluster API versions that will support the new Kubernetes release.
```
I think?

File: .github/ISSUE_TEMPLATE/kubernetes_bump.md [hunk 4]
```diff
@@ -26,24 +26,33 @@ changes should be cherry-picked to all release series that will support the new
...
-    * periodics & presubmits: 
-      * Bump `KUBEBUILDER_ENVTEST_KUBERNETES_VERSION` of the `test-mink8s` jobs to the new minimum supported management cluster version.
     * periodics:
       * Bump `KUBERNETES_VERSION_MANAGEMENT` of the `e2e-mink8s` job to the new minimum supported management cluster version.
       * Drop the oldest upgrade job as the oldest Kubernetes minor version is now out of support.
-  * Prior art: https://github.com/kubernetes/test-infra/pull/27421
+      * Add new upgrade job which upgrades from the previous to the new Kubernetes version.
+    * periodics & presubmits:
+      * Bump `KUBEBUILDER_ENVTEST_KUBERNETES_VERSION` of the `test-mink8s` jobs to the new minimum supported management cluster version.
+      * Adjust the `-latest` upgrade job to upgrade from the new Kubernetes and upgrade to the next Kubernetes version.
```
Comment (sbueringer) [inline .github/ISSUE_TEMPLATE/kubernetes_bump.md:44]:
```suggestion
      * Adjust the `-latest` upgrade job to upgrade from the new Kubernetes to the next Kubernetes version.
```

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks

/lgtm

File: .github/ISSUE_TEMPLATE/kubernetes_bump.md [hunk 5]
```diff
@@ -26,24 +26,33 @@ changes should be cherry-picked to all release series that will support the new
...
+  * Prior art: #9160
 * [ ] Ensure the jobs are adjusted to provide test coverage according to our [support policy](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions):
-  * For the main branch and the release branch of the latest supported Cluster API minor release:
-    * Add new periodic upgrade job.
-    * Adjust presubmit jobs so that we have the latest upgrade jobs available on PRs.
   * For the main branch:
-    * periodics & presubmits: 
-      * Bump `KUBEBUILDER_ENVTEST_KUBERNETES_VERSION` of the `test-mink8s` jobs to the new minimum supported management cluster version.
     * periodics:
       * Bump `KUBERNETES_VERSION_MANAGEMENT` of the `e2e-mink8s` job to the new minimum supported management cluster version.
```
Comment (sbueringer) [inline .github/ISSUE_TEMPLATE/kubernetes_bump.md:39]: I think this is also for the presubmit

Comment (chrischdi) [reply .github/ISSUE_TEMPLATE/kubernetes_bump.md:39]: Great point. I only did it for the periodics 😭

**Comment by fabriziopandini** [comment]
Thanks for updating this doc!
/lgtm

**Comment by sbueringer** [comment]
Thx

/lgtm

**Pull request merged**